source/drivers/MicroBitFile.cpp

```diff
@@ -0,0 +1,280 @@
+#include "MicroBitFile.h"
+#include "ErrorNo.h"
+
+/**
+  * Creates an instance of a MicroBitFile and creates a new file if required.
+  *
+  * @param fileName the name of the file to create/open.
+  *
+  * @param mode One of: READ, WRITE, READ_AND_WRITE. Defaults to READ_AND_WRITE.
+  */
+MicroBitFile::MicroBitFile(ManagedString fileName, int mode)
+{
+    this->fileName = fileName;
+
+    MicroBitFileSystem* fs;
+
+    if(MicroBitFileSystem::defaultFileSystem == NULL)
+        fs = new MicroBitFileSystem();
+    else
+        fs = MicroBitFileSystem::defaultFileSystem;
+
+    fileHandle = fs->open(fileName.toCharArray(), mode | CREATE);
```

Comment:
@remay `fs->open` will return an invalid file handle, so subsequent operations will fail on the instance.

I don't think the `CREATE` behaviour has any horrible side affects, although I agree that it is a bit weird to create a file if you only want to read it, so that should change. Also it should be noted that `mode` defaults to `READ_AND_WRITE` which is suitable for the general audience who will be using this class over `MicroBitFileSystem`, I think the `CREATE` behaviour should remain implicit with the exception of `READ`. 

Two potential solutions:
* Have an exception case for `READ`.
* Change the default of `mode` to be `READ_AND_WRITE | CREATE`.

I think I prefer the second solution, that way any modifications to that `mode` parameter will be as the user expects.